src/extension.ts

```diff
@@ -83,20 +83,43 @@ export function deactivate() { }
 
 const g_onSetLanguageClient = new vscode.EventEmitter<vslc.LanguageClient>()
 export const onSetLanguageClient = g_onSetLanguageClient.event
-function setLanguageClient(languageClient: vslc.LanguageClient = null) {
+function setLanguageClient(languageClient: vslc.LanguageClient = undefined) {
     g_onSetLanguageClient.fire(languageClient)
     g_languageClient = languageClient
 }
+function isLanguageClientActive() {
+    return g_languageClient !== undefined
+}
+
+export async function withLanguageClient(
+    callback: (languageClient: vslc.LanguageClient) => any,
+    callbackOnHandledErr: (err: Error) => any
+) {
+    if (!isLanguageClientActive()) {
+        return callbackOnHandledErr(new Error('Language client is not active'))
+    }
+    try {
+        return callback(g_languageClient)
+    } catch (err) {
+        if (err.message === 'Language client is not ready yet') {
+            return callbackOnHandledErr(err)
+        }
+        throw err
+    }
+}
 
 const g_onDidChangeConfig = new vscode.EventEmitter<vscode.ConfigurationChangeEvent>()
 export const onDidChangeConfig = g_onDidChangeConfig.event
 function changeConfig(event: vscode.ConfigurationChangeEvent) {
     g_onDidChangeConfig.fire(event)
     if (event.affectsConfiguration('julia.executablePath')) {
-        if (g_languageClient !== null) {
-            g_languageClient.stop()
-            setLanguageClient()
-        }
+        withLanguageClient(
+            languageClient => {
+                languageClient.stop()
+                setLanguageClient()
+            },
+            err => { }
+        )

```

Comment:
Uh, I think we should stay away from this callback style, isn't the whole idea of promises, `async` and `await` to avoid that? Also, that seems an unrelated design change, if we were to introduce it, can we do it in a different PR?

Reply:
Yeah we should modify the existing part in another PR. I will revert back this code.